src/libutil/experimental-features.cc

```diff
@@ -5,18 +5,61 @@
 
 namespace nix {
 
-std::map<ExperimentalFeature, std::string> stringifiedXpFeatures = {
-    { Xp::CaDerivations, "ca-derivations" },
-    { Xp::ImpureDerivations, "impure-derivations" },
-    { Xp::Flakes, "flakes" },
-    { Xp::NixCommand, "nix-command" },
-    { Xp::RecursiveNix, "recursive-nix" },
-    { Xp::NoUrlLiterals, "no-url-literals" },
-    { Xp::FetchClosure, "fetch-closure" },
-    { Xp::ReplFlake, "repl-flake" },
-    { Xp::AutoAllocateUids, "auto-allocate-uids" },
-    { Xp::Cgroups, "cgroups" },
-    { Xp::DiscardReferences, "discard-references" },
+std::map<ExperimentalFeature, std::pair<std::string, std::string>> stringifiedXpFeatures = {
```

Comment:
Let's create a struct for this, and use designed initializers down below. See e.g. the libexpr primops code for how that should be formatted.

Reply:
In fact, I might make this more "relational-DB-esque" in the sense of:

1. Have a vector of structs with all 3 fields.
2. Programmatically compute both indices:
    - `std::map<ExperimentalFeature, const FullExperimentalFeature &>`
    - `std::map<std::string, const FullExperimentalFeature &>`
 3. Assert during index building if there are any collisions.

N.B. `FullExperimentalFeature` is the name I just made up for the full row struct, but feel free to think of a different one.